pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -523,3 +525,80 @@ func ringBufferMemoryUsage(mapStats *model.EBPFMapStats, info *ebpf.MapInfo, k *
 	mapStats.RSS = mapStats.MaxSize
 	return nil
 }
+
+func hashMapNumberOfEntries(mp *ebpf.Map) int64 {
+	if isPerCPU(mp.Type()) {
+		return -1
+	}
+
+	numElements := 0
+	if ddebpf.BatchAPISupported() && mp.Type() != ebpf.HashOfMaps { // HashOfMaps doesn't work with batch API
```

Comment:
The ebpf check already has a minimum required kernel version. If it is higher than the one where batch lookups become available maybe we can simplify the code and always use batching.

Reply:
IIRC the batch API is introduced in 5.6 and the minimum kernel version is 5.5. There are also some inconsistencies in doing detection with versions, I think this feature has been backported to some old kernels, and isn't actually available in others (not sure about this, it's what I recall from the initial tests of the batch API).